File: programmers/Hash/1_maraton.py
# 완주하지 못한 선수
# 레벨 1
# 해쉬

def solution(participant, completion):
	n = len(completion)
	participant.sort()
	completion.sort()
	
	for i in range(0, n):
		if participant[i] != completion[i]:
			answer = participant[i]
			return answer

	answer = participant[-1]
	return answer

participant = ["mislav", "stanko", "mislav", "ana"]
completion = ["stanko", "ana", "mislav"]

solution(participant, completion)

File: programmers/Hash/1_maraton.js
/*
    완주하지 못한 선수
    레벨 1
    해쉬
*/

/* participant : 참여한 선수 이름이 담긴 배열, completion : 완주한 선수 이름이 담긴 배열 */
/* return : 완주하지 못한 선수 */
function solution(participant, completion) {
    const result = participant.filter(person => {
        if (completion.indexOf(person) === -1) {
            participant.
        }
        return completion.indexOf(person) === -1
    })
    return result[0];
}

const participant = ["mislav", "stanko", "mislav", "ana"]
const completion = ["stanko", "ana", "mislav"]

solution(participant, completion)